### DISCUSSION TITLE
Issues in calculating E_hull of matrials containing specific elements

### DISCUSSION THREAD

**Discussion by RyotaroOKabe** [janosh/matbench-discovery#97] [Q&A] [answered]
I have difficulty in predicting energy above hull (`E_hull`) ad formation energy (`E_form`) using `PatchedPhaseDiagram`. After I defined ppd_mp (https://github.com/janosh/matbench-discovery/blob/c9dc8ee6623ed121bc671fa1bffb9219ae8ba532/data/wbm/compile_wbm_test_set.py#L543), I have tried calculating E_hull and E_form using the line below: 

```py
e_above_hull = ppd_mp.get_e_above_hull(cse, allow_negative=True)
```

The error message is:
```py
ValueError: Unable to get decomposition for wbm-1-46 ComputedStructureEntry - La1 Yb1 Ag2  (LaYbAg2)
```

I also got an error from the line below: 
```py
e_form_ppd = ppd_mp.get_form_energy_per_atom(cse) - cse.correction_per_atom
```

The error message is:
```py
KeyError: Element Yb
```

As much as I surveyed, `pymatgen`'s phase diagram object cannot handle materials with elements like Yb and Rn. Is there any way we can use PatchedPhaseDiagram for all materials? 

I used the following versions for this work:
```
Python 3.11.7
pymatgen==2024.2.23
```

**Comment by CompRhys** [comment]
when you load/make the ppd it should have attributes `ppd.elements` and `ppd.el_refs` I can't immediately think of what might be causing your issues so if you could post the output of those two attributes it would be helpful to explore this futher.

**Reply by RyotaroOKabe** [reply]
[ppd_mp.el_refs.txt](https://github.com/janosh/matbench-discovery/files/14547413/ppd_mp.el_refs.txt)
[ppd_mp.elements.txt](https://github.com/janosh/matbench-discovery/files/14547414/ppd_mp.elements.txt)
Here I attached the outputs of `ppd_mp.elements` and `ppd_mp.el_refs` as txt files. I followed the code **build_phase_diagram.py** and **compile_wbm_test_set.py** for building PatchedPhaseDiagram ppd_mp.  Seems both of the outputs did not present Yb.

**Reply by janosh** [reply, marked as answer]
if you rerun the code that fetches all MP computes entries from the latest MP DB release (v2023.11.1), you won't get any `Yb` compounds. [they were all deprecated](https://docs.materialsproject.org/changes/database-versions#v2023.11.1) due to a poor choice of `Yb_2` pseudo-potential (see https://github.com/materialsproject/pymatgen/pull/2969 for details). i'd recommend you use the [compressed entries that were archived specifically for Matbench Discovery](https://figshare.com/articles/dataset/22715158?file=40344436).

**Reply by janosh** [reply]
@esoteric-ephemera is planning to rerun all `Yb`-containing MP compounds with `Yb_3` which when completed would allow you to build a new complete `PatchedPhaseDiagram`

**Reply by RyotaroOKabe** [reply]
Thanks. _ checked the archived data of Matbench Discovery v1.0.0 in the link you shared. So far, I could find only **2023-02-07-mp-computed-structure-entries.json.gz** as the most up-to-date dataset of Entry objects. Should I refer to any other data?
